File: api/src/main/java/com/jakedelivery/api/userorder/dto/UserOrderResponse.java
package com.jakedelivery.api.userorder.dto;

import com.jakedelivery.db._common.constant.UserOrderStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserOrderResponse {
    private Long id;
    private UserOrderStatus status;
    private BigDecimal amount;
    private LocalDateTime orderedAt;
    private LocalDateTime acceptedAt;
    private LocalDateTime cookingStartedAt;
    private LocalDateTime deliveryStartedAt;
    private LocalDateTime receivedAt;
}

File: api/src/main/kotlin/com/jakedelivery/api/userorder/dto/UserOrderResponse.kt
package com.jakedelivery.api.userorder.dto

import com.jakedelivery.db._common.constant.UserOrderStatus
import java.math.BigDecimal
import java.time.LocalDateTime

data class UserOrderResponse (
    val id: Long? = null,
    val status: UserOrderStatus? = null,
    val amount: BigDecimal? = null,
    val orderedAt: LocalDateTime? = null,
    val acceptedAt: LocalDateTime? = null,
    val cookingStartedAt: LocalDateTime? = null,
    val deliveryStartedAt: LocalDateTime? = null,
    val receivedAt: LocalDateTime? = null
)